content/developer/reference/frontend/services.rst

```diff
@@ -455,6 +457,100 @@ A notification that closes after a second:
   const close = notificationService.add("I will be quickly closed");
   setTimeout(close, 1000);
 
+.. _frontend/services/router:
+
+Router Service
+--------------
+
+Overview
+~~~~~~~~
+
+- Technical name: `router`
+- Dependencies: none
+
+The `router` service provides three features:
+
+* information about the current route
+* provides a way for the application to update the url, depending on its state
+* listens to every hash change, and notifies the rest of the application
+
+API
+~~~
+
+.. js:data:: current
+   :noindex:
+
+   The current route can be accessed with the ``current`` key. It is an object
+   with the following information:
+ 
+   * `pathname (string)` : the path for the current location (most likely `/web` )
+   * `search (object)` : a dictionary mapping each search keyword from the url to
+     its value. An empty string is the value if no value was explicitely given
+   * `hash (object)` : same as above, but for values described in the hash.
+
+For example:
+
+.. code-block:: javascript 
+
+  // url = /web?debug=assets#action=123&owl&menu_id=174
+  const { pathname, search, hash } = env.services.router.current;
+  console.log(pathname); //   /web
+  console.log(search); //   { debug="assets" }
+  console.log(hash); //   { action:123, owl: "", menu_id: 174 }
+
+
+Updating the URL is done with the  `pushState` method:
+
+.. js:function:: pushState(hash: object[, replace?: boolean])
+
+  :param Object hash: object containing a mapping from some key to some values
+  :param boolean replace: if true, the url will be replaced, otherwise key/value
+    pairs from the `hash` are updated.
+
+  Updates the URL with each key/value pair from the `hash` object. If a value is
+  set to an empty string, the key will be simply added to the url without any
+  corresponding value.
+
+  If true, the `replace` argument tells the router that the url hash should be
+  completely replaced (so, values not present in the `hash` object will be removed).
+
+  This method call does not reload the page. It only updates the url.
+
+For example:
+
+.. code-block:: javascript
+
+  // url = /web#action_id=123
+  routerService.pushState({ menu_id: 321 });
+  // url is now /web#action_id=123&menu_id=321
+  routerService.pushState({ yipyip: "" }, replace: true);
+  // url is now /web#yipyip
+
+Note that using `pushState` does not trigger a `hashchange` event, nor a
+`ROUTE_CHANGE` in the main bus. This is because this method is intended to be
+used "from the inside", to update the url so that it matches the actual current
+displayed state.
+
+Finally, the `redirect` method will redirect the browser to a specified url:
+
+.. js:function:: redirect(url[, wait])
+  
+  :param string url: a valid url
+  :param boolean wait: if true, wait for the server to be ready, and redirect after
```

Comment:
Maybe explain why we would need the server to be ready ? I have no clue what would be a valid use case